src/ssh_config/parser.rs

```diff
@@ -59,8 +59,8 @@ impl Parser {
 
     fn parse_raw(&self, reader: &mut impl BufRead) -> Result<(Host, Vec<Host>), ParseError> {
         let mut global_host = Host::new(Vec::new());
-        let mut is_in_host_block = false;
         let mut hosts = Vec::new();
+        let mut current_host: Option<Host> = None;
```

Comment:
```suggestion
        let mut current_host = Host::new(vec!["*".to_string()]);
```

`current_host` is the global host by default. What about this?